### PR TITLE
Handle operationalHoursSpecialInstructions field for VA health

### PR DESCRIPTION
**Pull request by brangi**
## Description
issue:  https://github.com/department-of-veterans-affairs/va.gov-team/issues/19615

## Testing done
Unit test

## Screenshots
Screen Shot 2021-02-10 at 2 41 13 PM

## Acceptance criteria

- [x] When flag switched for production, the language displayed on **VA health detail page** matches language from `operationalHoursSpecialInstructions` field  - https://github.com/department-of-veterans-affairs/vets-api/pull/5903
- [x] If null, nothing displays

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by brangi** [comment]
Flag added - https://github.com/department-of-veterans-affairs/vets-api/pull/5903

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    expect(wrapper.type()).to.not.equal(null);
+    expect(
+      wrapper
+        .render()
+        .find('#operational-special-p')
+        .text(),
+    ).to.equal(operationalHoursSpecialInstructions);
+    wrapper.unmount();
+  });
+});
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:53]: Can we add another test to confirm that the special instructions do NOT appear when:

- `showHoursSpecialInstructions` is `false
- `operationalHoursSpecialInstructions` is null
- facilityType !== `FacilityType.VA_HEALTH_FACILITY`

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        .text(),
+    ).to.equal(operationalHoursSpecialInstructions);
+    wrapper.unmount();
+  });
+
+  it('Should render LocationHours without operationalHoursSpecialInstructions field - showHoursSpecialInstructions false', () => {
+    const wrapper = shallow(
+      <LocationHours
+        location={testFacility}
+        showHoursSpecialInstructions={!operationalHoursSpecialInstructions}
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:58]: You could just pass `false` here I think.

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      operationalHoursSpecialInstructions:
+        'Administrative hours are Monday-Friday 8:00 a.m. to 4:30 p.m. | Expanded or Nontraditional hours are available for some services on a routine and or requested basis. Please call our main phone number for details. |',
+      uniqueId: '549',
+      visn: '17',
+      website: 'https://www.northtexas.va.gov/locations/directions.asp',
+    },
+  };
+
+  const { operationalHoursSpecialInstructions } = testFacility.attributes;
+  const showHoursSpecialInstructionsFlagTrue = true;
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:36]: You don't need this variable. To pass "true" to a react componen in JSXt, just leave out the argument, e.g.

```jsx
<LocationHours
  location={testFacility}
  showHoursSpecialInstructions
/>
```

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    expect(
+      wrapper
+        .render()
+        .find('#operational-special-p')
+        .text(),
+    ).to.equal(operationalHoursSpecialInstructions);
+    wrapper.unmount();
+  });
+
+  it('Should render LocationHours without operationalHoursSpecialInstructions field - showHoursSpecialInstructions false', () => {
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:55]: Shouldn't this be "Should NOT render..."?

Comment (mpelzsherman) [reply src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:55]: Same for the other 2 tests ...

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  };
+
+  const { operationalHoursSpecialInstructions } = testFacility.attributes;
+
+  it('Should render LocationHours with operationalHoursSpecialInstructions field with truthy values', () => {
+    const wrapper = shallow(
+      <LocationHours
+        location={testFacility}
+        /* eslint-disable react/jsx-boolean-value */
+        showHoursSpecialInstructions={true}
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:42]: This is passing eslint for me and doesn’t change the test results:

```jsx
<LocationHours location={testFacility} showHoursSpecialInstructions />
```

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      },
+      operationalHoursSpecialInstructions:
+        'Administrative hours are Monday-Friday 8:00 a.m. to 4:30 p.m. | Expanded or Nontraditional hours are available for some services on a routine and or requested basis. Please call our main phone number for details. |',
+      uniqueId: '549',
+      visn: '17',
+      website: 'https://www.northtexas.va.gov/locations/directions.asp',
+    },
+  };
+
+  const { operationalHoursSpecialInstructions } = testFacility.attributes;
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:35]: I’m not sure why yet but when I change it to this, the tests all pass:

```js
const operationalHoursSpecialInstructions = testFacility.attributes.operationalHoursSpecialInstructions;
```

File: src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        location={testFacility}
+        showHoursSpecialInstructions={false}
+      />,
+    );
+    expect(wrapper.find('#operational-special-p').length).to.equal(0);
+    wrapper.unmount();
+  });
+
+  it('Should render LocationHours without operationalHoursSpecialInstructions field - operationalHoursSpecialInstructions null', () => {
+    testFacility.attributes.operationalHoursSpecialInstructions = null;
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/components/LocationHours.unit.spec.jsx:67]: Here, you’re modifying the test data for subsequent tests.

I’d suggest making `testFacility` a function that takes an arg for `operationalHoursSpecialInstructions`.

**Review by mpelzsherman — APPROVED** [review]

**Review by jhonny218 — APPROVED** [review]
LGTM

**Pull request merged**